web/src/components/InfiniteScrollList.tsx

```diff
@@ -62,14 +63,14 @@ const InfiniteScrollList = <T,>({
   }, [page, hasMore, itemsPerPage, loadPage])
 
   useEffect(() => {
-    if (resetOnLanguageChange) {
+    if (resetOnLanguageChange || filterValues) {
       setData([])
       setError(null)
       setLoading(false)
       setHasMore(true)
       setPage(defaultPage)
     }
-  }, [languageCode, resetOnLanguageChange, defaultPage])
+  }, [languageCode, resetOnLanguageChange, defaultPage, filterValues])
```

Comment:
This works and is way more future proof/easier to understand:
``` typescript
  useEffect(() => () => {
    setData([])
    setError(null)
    setLoading(false)
    setHasMore(true)
    setPage(defaultPage)
  }, [loadPage, defaultPage])
```

Reply:
fixed